include/deal.II/base/config.h.in

```diff
@@ -116,7 +116,21 @@
  */
 
 #cmakedefine DEAL_II_WORDS_BIGENDIAN
-#define DEAL_II_COMPILER_VECTORIZATION_LEVEL @DEAL_II_COMPILER_VECTORIZATION_LEVEL@
+#define DEAL_II_HIGHEST_N_VECTORIZATION_BITS @DEAL_II_HIGHEST_N_VECTORIZATION_BITS@
+
+/*
+ * Backward compatibility setting
+ */
+#if DEAL_II_HIGHEST_N_VECTORIZATION_BITS == 512
+#define DEAL_II_COMPILER_VECTORIZATION_LEVEL 3
+#elif DEAL_II_HIGHEST_N_VECTORIZATION_BITS == 256
+#define DEAL_II_COMPILER_VECTORIZATION_LEVEL 2
+#elif DEAL_II_HIGHEST_N_VECTORIZATION_BITS == 128
+#define DEAL_II_COMPILER_VECTORIZATION_LEVEL 1
+#else
+#define DEAL_II_COMPILER_VECTORIZATION_LEVEL 0
+#endif
```

Comment:
Would you mind to test quickly whether it is possible to write something like
```
#if  DEAL_II_COMPILER_HAS_DIAGNOSTIC_PRAGMA
  // ...
  # define DEAL_II_COMPILER_VECTORIZATION_LEVEL _Pragma ("GCC warning \"The DEAL_II_COMPILER_VECTORIZATION_LEVEL macro is deprecated\"") 3
  // ...
#else 
  // what you have  
#endif 
```

Reply:
I will try that now.

Reply:
Works like a charm, I will add this together with the renaming of the variable.